storybookjs/storybook
# Sandboxes: Use Node 18 to generate Angular prerelease sandboxes

## What I did

This PR fixes the issue with Angular prereleases not working in our CI anymore, because they require Node 18.

- We now generate sandboxes both with Node 16 and Node 18. Only the Angular prerelease is on the Node 18 path, but it's set up so it's trivial to move other sandboxes over later.
- Merged the generate sandbox workflows for `main` and `next` into one, using the strategy matrix feature instead.
- Extracted Angular prerelease sandbox out of the regular CircleCI flow and into it's own separate that runs on Node 18. This is can also be duplicated later for other sandboxes, it's a bit hacky though. 



## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

1. See daily CI run for this PR: https://app.circleci.com/pipelines/github/storybookjs/storybook/58601/workflows/d3394727-854d-45b6-805c-c618fe147038
2. Merge this and trigger the sandbox generation flow. It can't be triggered until it's on the `next` branch because of the workflow rename, but here's a (cancelled) run where I log that the matrixes are set up correct: https://github.com/storybookjs/storybook/actions/runs/6037790919/job/16382747584

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_



### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
